test/models/work_test.rb

```diff
@@ -0,0 +1,287 @@
+require "test_helper"
+
+describe Work do
+  
+  let (:yml) { let_yml_superhash }
+  
+  describe "RELATIONS" do
+    
+    let(:movie1) { yml[:movie1] }
+    
+    it "can have many votes" do
+      assert(movie1.votes.count == 4)
+      movie1.votes.each do |vote|
+        assert(vote.class == Vote)
+        assert(vote.work_id == movie1.id)
+      end
+    end
+    
+    it "can add votes" do
+      new_vote = Vote.create(work: movie1, user_id: yml[:user6].id)
+      assert(movie1.votes.count == 5)
+    end
+    
+    it "work_obj.votes exists and can return Vote objs" do
+      movie1.votes.each do |vote|
+        assert(vote.class == Vote)
+        assert(vote.work.title == movie1.title)
+      end 
+    end
+    
+    it "if work_obj gets deleted, its dependent votes are too" do
+      assert(movie1.votes.count == 4)
+      
+      movie1.destroy 
+      refute(Work.find_by(title: "joker"))
+      
+      assert(movie1.votes.count == 0)
+      # what happens to the votes themselves? We're testing that in vote_test.rb
+    end
+    
+  end
+  
+  describe "VALIDATIONS" do
+    it "Can create Work obj with correct attributes" do
+      categories = %w[movie album book]
+      titles = %w[aaa bbb ccc]
+      pub_years = [123, 456, 789]
+      creators = %w[me you they]
+      descriptions = %w[xxx yyy zzz]
+      categories.each_with_index do |category, index|
+        Work.create!(category: categories[index], 
+          title: titles[index], 
+          published_year: pub_years[index], 
+          creator: creators[index], 
+          description: descriptions[index]
+        )
+        new_work = Work.last
+        
+        assert(new_work.category == categories[index])
+        assert(new_work.title == titles[index])
+        assert(new_work.published_year == pub_years[index])
+        assert(new_work.votes == [])
+        assert(new_work.creator == creators[index])
+        assert(new_work.description == descriptions[index])
```

Comment:
This is assert-style testing.  It works, but our style of testing (spec-style) looks more like:

```ruby
expect(new_work.category).must_equal categories[index]
```